compiler/hash-typecheck/src/new/passes/scope_discovery.rs

```diff
@@ -443,20 +459,32 @@ impl<'tc> ScopeDiscoveryPass<'tc> {
                     "Found module pattern in stack definition"
                 )
             }
-            ast::Pat::Tuple(tuple_pat) => {
-                for entry in tuple_pat.fields.ast_ref_iter() {
+            ast::Pat::Tuple(ast::TuplePat { fields, spread }) => {
+                for entry in fields.ast_ref_iter() {
                     self.add_stack_members_in_pat_to_buf(entry.pat.ast_ref(), buf);
                 }
+
+                if let Some(spread) = spread {
+                    register_spread_pat(spread, buf);
+                }
             }
-            ast::Pat::Constructor(constructor_pat) => {
-                for field in constructor_pat.fields.ast_ref_iter() {
+            ast::Pat::Constructor(ast::ConstructorPat { fields, spread, .. }) => {
+                for field in fields.ast_ref_iter() {
                     self.add_stack_members_in_pat_to_buf(field.pat.ast_ref(), buf);
                 }
+
+                if let Some(spread) = spread {
+                    register_spread_pat(spread, buf);
+                }
             }
-            ast::Pat::List(list_pat) => {
-                for pat in list_pat.fields.ast_ref_iter() {
+            ast::Pat::List(ast::ListPat { fields, spread }) => {
+                for pat in fields.ast_ref_iter() {
                     self.add_stack_members_in_pat_to_buf(pat, buf);
                 }
+
+                if let Some(spread) = spread {
+                    register_spread_pat(spread, buf);
+                }
```

Comment:
Ideally the register_spread_pat call should happen at the iteration corresponding to the position of the spread in the pattern list.